### DISCUSSION TITLE
How to generate CDR serialization vs json

### DISCUSSION THREAD

**Discussion by sbrett9** [jmmorato/openddsharp#256] [General]
Howdy, I've been trying to get the opensharpdds nuget plugin to work.
It works great, really neat capability.
I am however unable to find a way to serialize the data to a native in a non-json fashion.
Specifically I have a need to access the raw native bytes in managed space prior to sending them off to the DataWriter.
It looks as if there is template for generating code with this capability, the csharp_cdr_generator, but I can't figure out what I need to do in the project generation step, see _dotnet new openddsharp-idl-project --name TestMessage --output TestMessage_ for example, to active the cdr generator output instead of the csharp_json  output?

**Comment by jmmorato** [comment]
CDR serialization is a feature I'm still working on, therefore not release yet.
[This is the project](https://github.com/jmmorato/openddsharp/blob/39563ac15ee1f77b1a53bd7f315597e654452df9/Tests/TestIdlCdr/TestIdlCdr.csproj) that I'm using for testing the serialization, you can create something similar for your solution but keep in mind it is still not fully finished.

**Comment by sbrett9** [comment]
Ah ha.
I did suspect it may be a WIP, given the tad bit of json references i saw
bleeding over into the CDR portion.
I must say its very cool stuff nontheless.

Thank you for your time and answer.
